tests/AbstractLoopTest.php

```diff
@@ -16,11 +16,39 @@ public function setUp()
 
     abstract public function createLoop();
 
+    /**
+     * @deprecated Use createSocketPair() instead.
+     */
     public function createStream()
     {
         return fopen('php://temp', 'r+');
     }
+    
+    public function isUnixSocketSupported()
+    {
+        return in_array('unix', stream_get_transports(), true);
+    }
 
+    public function createSocketPair()
+    {
+        if ($this->isUnixSocketSupported()) {
+            $sockets = stream_socket_pair(STREAM_PF_UNIX, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
+        } else {
+            $sockets = stream_socket_pair(STREAM_PF_INET, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
+        }
+        
+        foreach ($sockets as $socket) {
+            if (function_exists('stream_set_read_buffer')) {
```

Comment:
Disabling the read buffers is necessary because some event loop extensions will not report streams as readable if PHP does read buffering. Setting the buffer to 0 will ensure that all backends will work on the socket directly.